deploy/deploy_test.go

```diff
@@ -442,14 +549,18 @@ func TestMetalLBSpec(t *testing.T) {
 		m: &MetalLBSpec{
 			IPCount: 20,
 		},
-		execer: exec.NewFakeExecer(errors.New("namespace error")),
-		dErr:   "namespace error",
+		resp: []fexec.Response{
+			{Cmd: "kubectl", Args: []string{"apply", "-f", ""}, Err: "namespace error"},
+		},
+		dErr: "namespace error",
 	}, {
 		desc: "secret create",
+		resp: []fexec.Response{
+			{Cmd: "kubectl", Args: []string{"apply", "-f", ""}, Err: "secret error"},
```

Comment:
this shouldn't need to error

Reply:
The test hangs without it.  Needs more investigation.